### PR TITLE
[Feature] Enabling Route Props - Menu Display

### PR DESCRIPTION
**Pull request by tarique93102**

### CODE REVIEW

File: resumebuilder.client/src/Pages/App.tsx [hunk 1]
```diff
@@ -1,27 +1,78 @@
...
+        super(props);
+
+        this.state = {
+            DisplayMenu: true
+        };
+
+        this.handleDisplayMenu = this.handleDisplayMenu.bind(this);
+    }
+
+    private handleDisplayMenu(value: boolean) {
```
Comment (yrshaikh) [inline resumebuilder.client/src/Pages/App.tsx]: Better to change this rename to `hideHeader()` and not accept a bool. By default you are already setting DisplayMenu to true.

File: resumebuilder.client/src/Pages/App.tsx [hunk 2]
```diff
@@ -1,27 +1,78 @@
...
+};
+
+class App extends React.Component<IProps, IState> {
+    state: IState;
+
+    constructor(props: IProps) {
+        super(props);
+
+        this.state = {
+            DisplayMenu: true
```
Comment (yrshaikh) [inline resumebuilder.client/src/Pages/App.tsx]: `EnableHeader` or `ShowHeader` please. Camelcase.

File: resumebuilder.client/src/Pages/App.tsx [hunk 3]
```diff
@@ -1,27 +1,78 @@
...
+
+        return (
+            <Layout className="AppLayout">
+                {(this.state.DisplayMenu) ? HeaderLayout : null}
+                <Content className="AppLayout__Content">
+                    <Switch>
+                        <Route
+                            exact
+                            path="/"
+                            render={(props: IProps) => <Home {...props} handleDisplayMenu={this.handleDisplayMenu} />}
```
Comment (yrshaikh) [inline resumebuilder.client/src/Pages/App.tsx]: Home and Feature component doesnt need this.

File: resumebuilder.client/src/Pages/Entry/Entry.tsx [hunk 4]
```diff
@@ -1,8 +1,17 @@
...
+    handleDisplayMenu: Function
+}
+
+class Entry extends React.Component<IEntryProps, any> {
+    componentDidMount() {
+        this.props.handleDisplayMenu(false);
+    }
+
     render() {
+        console.log(this.props);
```
Comment (yrshaikh) [inline resumebuilder.client/src/Pages/Entry/Entry.tsx]: Please remove.

File: resumebuilder.client/src/Pages/App.tsx [hunk 5]
```diff
@@ -1,27 +1,78 @@
...
+import Home from "./Home/Home";
+import Entry from "./Entry/Entry";
+import Features from "./Features/Features";
 
 const { Header, Content } = Layout;
 
-const App: React.FC = () => {
-    return (
-        <Layout className="AppLayout">
+interface IProps {
```
Comment (yrshaikh) [inline resumebuilder.client/src/Pages/App.tsx]: `IAppProps` and `IAppState`.

File: resumebuilder.client/src/Pages/App.tsx [hunk 6]
```diff
@@ -1,27 +1,78 @@
...
+};
+
+interface IState {
+    DisplayMenu: Boolean
```
Comment (yrshaikh) [inline resumebuilder.client/src/Pages/App.tsx]: use `boolean` primitive type instead.

File: resumebuilder.client/src/index.tsx [hunk 7]
```diff
@@ -3,8 +3,9 @@ import ReactDOM from 'react-dom';
 import './index.css';
 import App from './Pages/App';
 import * as serviceWorker from './serviceWorker';
+import {BrowserRouter as Router} from 'react-router-dom';
 
-ReactDOM.render(<App />, document.getElementById('root'));
+ReactDOM.render(<Router><App /></Router>, document.getElementById('root'));
```
Comment (yrshaikh) [inline resumebuilder.client/src/index.tsx]: 👍

File: resumebuilder.client/src/Pages/App.tsx (hunk 1 above)
Comment (tarique93102) [reply resumebuilder.client/src/Pages/App.tsx]: Yes. But I believe when we start navigating programmatically, we will need to reset the state of the pages requiring the header. I didn't get a chance to test it but that is one issue which might happen.

File: resumebuilder.client/src/Pages/App.tsx (hunk 3 above)
Comment (tarique93102) [reply resumebuilder.client/src/Pages/App.tsx]: As mentioned. Might need it. But I will still test it once to be absolutely sure.

**Pull request merged**